aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cpp

```diff
@@ -725,10 +720,11 @@ _scaled_dot_product_flash_attention_nestedtensor_cuda(
           return_debug_mask,
           scale);
   // Reshape output to convert nnz to batch_size and seq_len
+  // Don't reshape {q,k,v}_padded since they are only used for backwards
```

Comment:
Aside from this one comment, all the other changes seem cosmetic?

Reply:
you are correct, in fact this is an old comment. And I tweaked the behavior of the flash_attention slightly in order to make it conform to the autograd invariants. I left a comment in the summary but essentially all the padding happens in the top level sdpa where it will get automatically tracked by autograd.